### DISCUSSION TITLE
box에 대해 이야기 해주세요

### DISCUSSION THREAD

**Discussion by skyhanull** [wanted-9th-3team/frontend-study#50] [프론트 기술 질문 공간]
{ box-sizing: border-box; }는 무엇이고 장점은 무엇인가요?

**Comment by javascriptbiggosoo** [comment]
element에 padding을 조정해도 element의 크기가 변하지 않도록 해줍니다.
padding하나 건드렸다고 모든 UI가 깨지는 사태를 막아주기 때문에 저는 초기 설정에서 reset.css와 더불어 body에 box-sizing: border-box를 꼭 사용해줍니다.

**Comment by BaekYuri** [comment]
box-sizing: content-box인 경우에는, width:100px, padding:10px, border: 5px solid black로 줄 경우에, 넓이가 100+10+5+5= 120px가 됩니다. 이러한 경우처럼 아무리 넓이를 100px를 설정하여도 패딩이나 마진을 변경하는 경우에, 100px가 아닌 의도되지 않은 값이 나올 수 있습니다.
패딩과 마진에 관계없이 100px로 고정하고 싶은 경우에 border-box를 사용합니다. ui의 크기 변경으로 인해 의도치 않은 ui가 나오는 것을 방지할 수 있습니다.
